AddRoutineView.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  AddRoutineView.swift
+//  OnTrack
+//
+//  Allows user to add a new routine to their list of routines.
+//
+//  Created by Noah Sadir on 4/4/24.
+//
+//  Contributors: David
+//
+
+import SwiftUI
+
+struct AddRoutineView: View {
+    @State var id: String = ""
+    @State var title: String = ""
+    @State var description: String = ""
+    @State var category: String = ""
+    @State var frequency: Int = 0
+    @State var dates: String = ""
+    @State var streakLength: Int = 0
+    
+    @StateObject var vm = ViewModel()
+    
+    var body: some View {
+        VStack {
+            HStack {
+                Text("OnTrack")
+                    .font(.largeTitle)
+                    .bold()
+                    .padding(.leading)
+                
+                Spacer()
+                
+                
+            }
+            
+            Text("Add Routine")
+                .padding(.vertical)
+                .font(.largeTitle)
+               .italic()
+            
+            ZStack {
+                Rectangle()
+                    .foregroundStyle(Color(red: 0.45, green: 0.85, blue: 0.55))
+                
+                TextField("Add an ID", text: $id)
+                    .foregroundStyle(.black)
+                    .padding([.leading, .top])
+                    .frame(maxHeight: .infinity, alignment: .top)
+            }
+            .frame(width: 360, height: 20)
```

Comment:
The user does not specify an ID.

This is done by the app itself so that the object conforms to `Identifiable`.

There's no clear protocol for how to do this yet, just try to generate a random string for now.